lib/src/components/switch/zeta_switch.dart

```diff
@@ -0,0 +1,83 @@
+import 'dart:io';
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+
+import '../../../zeta_flutter.dart';
+import 'material_switch.dart';
+
+const _sizeAndroid = Size(48, 24);
+const _sizeIOS = Size(56, 32);
+
+/// Zeta Switch.
+///
+/// Switch can turn an option on or off.
+class ZetaSwitch extends StatelessWidget {
+  /// Constructor for [ZetaSwitch].
+  const ZetaSwitch({
+    super.key,
+    this.value = false,
+    this.onChanged,
+    this.size,
+  });
+
+  /// Constructor for [ZetaSwitch] for the platform Android.
+  const ZetaSwitch.android({
+    super.key,
+    this.value,
+    this.onChanged,
+  }) : size = _sizeAndroid;
+
+  /// Constructor for [ZetaSwitch] for the platform Android.
+  const ZetaSwitch.ios({
+    super.key,
+    this.value,
+    this.onChanged,
+  }) : size = _sizeIOS;
+
+  /// Determines whether the switch is on or off.
+  final bool? value;
+
+  /// Called when the value of the switch should change.
+  final ValueChanged<bool?>? onChanged;
+
+  /// Determines the size of the switch.
+  /// If not specified, the size will be determined
+  /// depending on the Platform and predefined constants.
+  final Size? size;
```

Comment:
Do we need a property for the size? From the figma it looks like the size should just be determined by the platform